nomos-services/chain-service/src/storage/mod.rs

```diff
@@ -67,3 +69,48 @@ pub trait StorageAdapter<RuntimeServiceId> {
         blocks: BTreeMap<Slot, HeaderId>,
     ) -> Result<(), overwatch::DynError>;
 }
+
+const LOG_TARGET: &str = "cryptarchia::service::storage";
+
+/// Extension trait for [`StorageAdapter`] that provides additional
+/// functionalities.
+#[async_trait::async_trait]
+pub trait StorageAdapterExt<RuntimeServiceId>: StorageAdapter<RuntimeServiceId> {
```

Comment:
what's the reason for an extension trait instead of adding these to the `StorageAdapter` trait?